api/src/main/java/eu/darkbot/api/managers/OreAPI.java

```diff
@@ -74,9 +87,35 @@ enum Ore {
             this(id, upgradable, true);
         }
 
+        public static Ore of(int id) {
+            for (Ore ore : VALUES) {
+                if (ore.getId() == id)
+                    return ore;
+            }
+            return null;
+        }
+
         public String getName() {
             return name().toLowerCase(Locale.ROOT);
         }
+    }
+
+    interface Upgrade {
+        Ore getOre();
+        int getAmount();
+    }
 
+    enum UpgradePlace {
```

Comment:
This needs a better name, maybe UpgradeType, UpgradableTrait, or something along those lines